### PR TITLE
[Docs] Adding generic legends

### PR DESCRIPTION
**Pull request by SAHU-01**
**Notes for Reviewers**

This PR fixes missing generic legends in documentation of Shapes

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 50e015b23ecdc45b68bd112fdaa4c9ca40a99123 at: https://66aa3bcd1207153a5aec4820--meshery-docs-preview.netlify.app

**Comment by SAHU-01** [comment]
The description and common use-cases related to each shape are yet to added in docs, it is currently present [here](https://docs.google.com/spreadsheets/d/14dXTKTqn_PB9E87xhanaIUIudiBpka54eqluxLxD028/edit?usp=sharing) and needs confirmation.

**Review by leecalcote — COMMENTED** [review]
Thanks!

The "SqlJob", "User" - are these shapes that Meshery has?

The "Textbox" icon is presented differently, if my memory serves.

It'd be great to get all icons uniformly presented of the same color.

**Review by leecalcote — COMMENTED** [review]
@vishalvivekm, I wonder if we wouldn't be better suited to use a datafile here. Thoughts?

See also: https://docs.google.com/spreadsheets/d/1nUgr0b7Ju8FxeWBLAgBO_OiYkaQkPWRh0gOX5M0cByc/edit

**Comment by SAHU-01** [comment]
> Thanks!
> 
> The "SqlJob", "User" - are these shapes that Meshery has?
> 
> The "Textbox" icon is presented differently, if my memory serves.
> 
> It'd be great to get all icons uniformly presented of the same color.

Yes, I found them here so I added these in the shapes listed on meshmap they are absent though:
- https://github.com/meshery/meshery/blob/master/server/meshmodel/meshery-shapes/0.7.2/v1.0.0/components/SqlJob.json
- https://github.com/meshery/meshery/blob/master/server/meshmodel/meshery-shapes/0.7.2/v1.0.0/components/User.json

Yes you are right the textbox and section look same just the fill color is different for both : I'll check the json again and fix this
image

currently for the newly added icons the color is based on what the json returns, for the previous ones I'll modify them to match the current color scheme

**Comment by l5io** [comment]
🚀 Preview for commit 6babbffece8f9c4c53e6c11dde634d042e04b121 at: https://66ace11d64f3fc5698377620--meshery-docs-preview.netlify.app

**Comment by vishalvivekm** [comment]
the data file (csv)  is now accessible throughout the site in the variable `site.data.shapes`

**Comment by leecalcote** [comment]
@vishalvivekm, I retract my suggestion that we'll need to have an ongoing refresh of this datafile. This isn't necessary. The spreadsheet can go away (soon) and the datafile take its place.

**Comment by vishalvivekm** [comment]
> @vishalvivekm, I retract my suggestion that we'll need to have an ongoing refresh of this datafile. This isn't necessary. The spreadsheet can go away (soon) and the datafile take its place.

Roger

**Comment by SAHU-01** [comment]
A couple of things we might want look into @vishalvivekm and @leecalcote 

1. For TextBox I found Namespace in figma would this be an ideal shape?
image

2. Or for textbox sticking to 

image

as represented in Meshmap design section is good?
3. For SQLJob I'm not sure what description would fit and where it is used.
4. The rectangle, section and TextBox in Meshmap design section all three have same shape but in figma we've got a separate rectangle and square for generic legends which shape do you suggest should be used?
image
5. Is it necessary to create another datafile ? [Here](https://github.com/meshery/meshery/tree/master/server/meshmodel/meshery-shapes/0.7.2/v1.0.0/components) there is an empty key for description and complete svg, can we not add it here?

File: docs/_data/shapes.csv [hunk 1]
```diff
@@ -0,0 +1,203 @@
+Shape,Description,Common usecase,Depend on,Note,SVG
```
Comment (leecalcote) [inline docs/_data/shapes.csv:1]: @vishalvivekm, @SAHU-1 in order to maintain this data file long-term, I propose converting to YAML.

File: docs/_data/shapes.csv [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+  </svg>"
+Trapezoid,"A four-sided shape with one pair of parallel sides, used to indicate hierarchical relationships or steps in a process.","- Representing funnels in process flows (e.g., sales funnels)
+- Indicating converging or diverging processes
+- Showing hierarchical structures with varying levels of detail",Dependency: Often used to show narrowing or widening of scope in processes,,"<svg viewBox=""0 0 40 40"" fill=""none"" xmlns=""http://www.w3.org/2000/svg"">
+<path d=""M8.5 3H32.5L39.5 37H1.5L8.5 3Z"" fill=""#00B39F"" fill-opacity=""0.1"" stroke=""#00B39F"" stroke-miterlimit=""10"" stroke-width=""1""/>
+</svg>"
+SqlJob,,,,,"<svg viewBox=""0 0 40 40"" fill=""none"" xmlns=""http://www.w3.org/2000/svg"">
```
Comment (leecalcote) [inline docs/_data/shapes.csv:170]: Comment out this shape given that it isn't included in releases yet, @sahu-01.

// @aabidsofi19

File: docs/_data/shapes.csv [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+<circle cx=""20"" cy=""20"" r=""19"" fill=""#00B39F"" fill-opacity=""0.1"" stroke=""#00B39F"" stroke-width=""2""/>
+<text x=""50%"" y=""50%"" dominant-baseline=""middle"" text-anchor=""middle"" font-family=""Arial, sans-serif"" font-size=""10"" fill=""#00B39F"">SQL</text>
+</svg>"
+User,"This node represents an individual user, often depicted with an icon or simple human silhouette.","- Representing end-users or user roles in system diagrams
```
Comment (leecalcote) [inline docs/_data/shapes.csv:174]: Comment out this shape given that it isn't included in releases yet, @sahu-01.

// @aabidsofi19

**Review by leecalcote — CHANGES_REQUESTED** [review]

**Comment by l5io** [comment]
🚀 Preview for commit f6841e2e50de591485ef3cfa5a15b293da0761c2 at: https://66b1b9def46ccd8531aeb2be--meshery-docs-preview.netlify.app

**Comment by SAHU-01** [comment]
@leecalcote this is what it looks like now:
image

I added the YAML file and mapped data from here in a tabular structure, things I'm unable to debug currently:

1. The common usecases and depends on though present in YAML return a null when I try to map them in the table
2. Despite commenting SQLJob and User in YAML it still shows up
3. The polygon and Octagon SVG shapes need to be fixed height sizes are too big.

**Comment by l5io** [comment]
🚀 Preview for commit 4a8b7455e91577be4f77cad7cbee9e8b241641ac at: https://66b1bfed871fef8b5049ec63--meshery-docs-preview.netlify.app

**Comment by leecalcote** [comment]
@Ashparshp, help, if you can.

@vishalvivekm you might have input to offer.

**Comment by Ashparshp** [comment]
@SAHU-01 let's discuss this PR during the Meshery dev meeting call, if you would. Please add it as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit?usp=drivesdk).

**Comment by vishalvivekm** [comment]
Thank you @SAHU-01 
We also would want to make the table responsive in mobile-view.
make it scrollable in smaller screen-width ?
Here's an example table: https://docs.meshery.io/concepts/architecture#meshery-connections-and-their-actions

Also, instead of having SVGs in the data file, I wonder if we could have a assets/shapes/<all the svgs here> and in the data file just the path to svgs.

**Comment by Ashparshp** [comment]
@SAHU-01 LMK if you need any help.

**Comment by SAHU-01** [comment]
Thanks @Ashparshp , I believe I can sort the image size issue if we remove those svgs and just add the SVG path info in datafiles!

**Comment by Muhammed-Moinuddin** [comment]
Screenshot from 2024-08-10 06-17-21
We should use space like we are doing with other names.
Can we display them in alphabetical order? [Shapes from A - Z]

**Comment by vishalvivekm** [comment]
Thank you @SAHU-01 for putting all the efforts.
Will you please push your latest changes ?

File: docs/pages/extensions/meshmap/Node_ Volume.svg
Comment (vishalvivekm) [inline docs/pages/extensions/meshmap/Node_ Volume.svg:1]: let's have url-friendly names for svg files i.e `node_volume.svg` :heavy_check_mark:  `node_ volume.svg` :x: 
@SAHU-01

File: docs/pages/extensions/meshmap/Polygon 6.svg
Comment (vishalvivekm) [inline docs/pages/extensions/meshmap/Polygon 6.svg:1]: same here too, let's not have whitespaces in the file names @SAHU-01

**Review by vishalvivekm — CHANGES_REQUESTED** [review]

File: docs/pages/extensions/meshmap/Node_ Volume.svg
Comment (SAHU-01) [reply docs/pages/extensions/meshmap/Node_ Volume.svg:1]: Heya, I did not add these I'll make sure to remove these files.

File: docs/pages/extensions/meshmap/Polygon 6.svg
Comment (SAHU-01) [reply docs/pages/extensions/meshmap/Polygon 6.svg:1]: Same goes for this one, I'll remove this as well.

**Comment by l5io** [comment]
🚀 Preview for commit ce1e6c75a9409f0abf59eab07fff790e5dae762b at: https://66bb64896652183f73b2eed1--meshery-docs-preview.netlify.app

File: docs/pages/extensions/meshmap/component-shape-guide.md [hunk 4]
```diff
@@ -13,55 +13,134 @@ category: meshmap
...
+      font-weight: bold;
+      white-space: nowrap;
+    }
+
+    td img {
+      width: 50px;
+      height: 50px;
+    }
+  }
+</style> -->
```
Comment (vishalvivekm) [inline docs/pages/extensions/meshmap/component-shape-guide.md:108]: should we get rid of the commented out styles @SAHU-01 ?

Comment (vishalvivekm) [inline docs/pages/extensions/meshmap/component-shape-guide.md:108]:
```suggestion
```

Comment (vishalvivekm) [reply docs/pages/extensions/meshmap/component-shape-guide.md:108]: @SAHU-01 ^^

Comment (SAHU-01) [reply docs/pages/extensions/meshmap/component-shape-guide.md:108]: I'll remove the commented portion

**Comment by l5io** [comment]
🚀 Preview for commit 91e2e6c38a8e420633bf0011f706a3d7d530c9c2 at: https://66bb6ac7ab60233d14659a3d--meshery-docs-preview.netlify.app

**Review by vishalvivekm — APPROVED** [review]
Thank you @SAHU-01 
I'll follow up with a PR.

**Pull request merged**